core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpHeaders.java

```diff
@@ -0,0 +1,199 @@
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.RequestHeadersBuilder;
+import com.linecorp.armeria.internal.common.ArmeriaHttpUtil.CaseInsensitiveMap;
+
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.util.AsciiString;
+
+public final class ArmeriaHttpHeaders extends HttpHeaders {
+
+    private final RequestHeadersBuilder builder;
+
+    public ArmeriaHttpHeaders(RequestHeadersBuilder builder) {
+        this.builder = builder;
+    }
+
+    public ArmeriaHttpHeaders(RequestHeadersBuilder builder, HttpHeaders headers) {
+        this.builder = builder;
+        headers.forEach(e -> builder.add(e.getKey(), e.getValue()));
+    }
+
+    @Override
+    public String get(String name) {
+        return builder.get(name);
+    }
+
+    @Override
+    public Integer getInt(CharSequence name) {
+        return builder.getInt(name);
+    }
+
+    @Override
+    public int getInt(CharSequence name, int defaultValue) {
+        return builder.getInt(name, defaultValue);
+    }
+
+    @Override
+    public Short getShort(CharSequence name) {
+        return builder.getInt(name).shortValue();
+    }
+
+    @Override
+    public short getShort(CharSequence name, short defaultValue) {
+        return (short) builder.getInt(name, defaultValue);
+    }
+
+    @Override
+    public Long getTimeMillis(CharSequence name) {
+        return builder.getTimeMillis(name);
+    }
+
+    @Override
+    public long getTimeMillis(CharSequence name, long defaultValue) {
+        return builder.getTimeMillis(name, defaultValue);
+    }
+
+    @Override
+    public List<String> getAll(String name) {
+        return builder.getAll(name);
+    }
+
+    @Override
+    public List<Entry<String, String>> entries() {
+        return builder.stream()
+                      .map(e -> Map.entry(e.getKey().toString(), e.getValue()))
+                      .collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean contains(String name) {
+        return builder.contains(name);
+    }
+
+    @Override
+    public Iterator<Entry<String, String>> iterator() {
+        return builder.stream()
+                      .map(e -> Map.entry(e.getKey().toString(), e.getValue()))
+                      .iterator();
+    }
+
+    @Override
+    public Iterator<Entry<CharSequence, CharSequence>> iteratorCharSequence() {
+        return  builder.stream()
+                       .map(e -> Map.entry((CharSequence) e.getKey().toString(),(CharSequence) e.getValue()))
+                       .iterator();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return builder.isEmpty();
+    }
+
+    @Override
+    public int size() {
+        return builder.size();
+    }
+
+    @Override
+    public Set<String> names() {
+        return builder.names().stream().map(AsciiString::toString).collect(Collectors.toSet());
+    }
+
+    @Override
+    public HttpHeaders add(String name, Object value) {
```

Comment:
Could you create a test for this class?
We can move the tests for converting Netty HTTP/1.1 to Armeria headers in `ArmeriaHttpUtilTest`

Reply:
Sure I will!